File: multi/infrastructure/src/main/java/com/example/infrastructure/theater/TheaterRepositoryImpl.java
package com.example.infrastructure.theater;

import com.example.domain.theater.Theater;
import com.example.domain.theater.service.TheaterRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Repository;

@Repository
@Slf4j
@RequiredArgsConstructor
public class TheaterRepositoryImpl implements TheaterRepository {

    private final TheaterJpaRepository theaterJpaRepository;

    @Override
    public Theater getTheaterInfo(Long theaterId) {
        return theaterJpaRepository.findById(theaterId).orElse(null);
    }
}

File: data/src/main/kotlin/kr/spartacodingclub/cinema/data/repository/TheaterRepositoryImpl.kt
package kr.spartacodingclub.cinema.data.repository

import kr.spartacodingclub.cinema.core.domain.Theater
import kr.spartacodingclub.cinema.core.repository.TheaterRepository
import kr.spartacodingclub.cinema.data.mapper.TheaterMapper
import org.springframework.stereotype.Repository

@Repository
class TheaterRepositoryImpl(
    private val theaterJpaRepository: TheaterJpaRepository
) : TheaterRepository {
    override fun findById(id: Long): Theater? {
        return theaterJpaRepository.findById(id)
            .map(TheaterMapper.INSTANCE::toTheater)
            .orElse(null)
    }
}
